plugins/flutter_aepedgeconsent/test/flutter_aepedgeconsent_test.dart

```diff
@@ -44,6 +44,68 @@ void main() {
       expect(await Consent.extensionVersion, testVersion);
     });
   });
-}
 
-//TO DO: Add tests
\ No newline at end of file
+  group('getConsents', () {
+    final Map<String, String> consentPreferenceStringMap = {
+      "val": "y",
+    };
+
+    final Map<String, dynamic> consentsMap = {
+      "collect": consentPreferenceStringMap,
+    };
+
+    final Map<String, dynamic> testGetConsent = {
+      "consents": consentsMap,
+    };
+
+    final List<MethodCall> log = <MethodCall>[];
+
+    setUp(() {
+      channel.setMockMethodCallHandler((MethodCall methodCall) async {
+        log.add(methodCall);
+        return testGetConsent;
+      });
+    });
+
+    test('invokes correct method', () async {
+      await Consent.consents;
+
+      expect(log, <Matcher>[
+        isMethodCall(
+          'getConsents',
+          arguments: null,
+        ),
+      ]);
+    });
+
+    test('returns correct result', () async {
+      expect(await Consent.consents, testGetConsent);
```

Comment:
nit: "testGetConsent" variable name looks like a test function name. It would be better to name it like `expectedConsentValue` , `actualConsentValue` etc..

Reply:
Ideally we should not mock the immediate function if we want to check for actual values. So here if you want to check for getConsents returning proper value, we should be mocking the functions that getConsents is using instead of mocking getConsents itself.